controls/views.py

```diff
@@ -817,26 +822,33 @@ def restore_to_history(request, smt_id, history_id):
     """
     Restore the current model instance to a previous version
     """
+    full_smt_history = None
+    try:
+        smt = Statement.objects.get(id=smt_id)
```

Comment:
Since none of the remaining code really makes sense if this line doesn't succeed (such as subsequent calls using the smt object), I'd put a `return` statement in the `except` clause of this try-except block to terminate execution of this view early, rather than trying to perform the rest of the processing with a null object. Maybe by either copy-pasting this line at the end of the `except`:
`return render(request, "controls/statement_history.html", context)` (and adding placeholders for the context attributes), or reworking the code flow structure somehow. But adding another return statement in the except is probably the lowest effort solution.
Granted, by this point you've already initialized all the variables you are passing out of the view to the template, so any errors in the `try` blocks should just pass the defaults to the template, but I'm talking more about wasting computing and lookup cycles once we already know the code will break. It's a theoretical performance improvement approach, but we can talk about it more outside.

Reply:
After the first error in a try it would fail when providing invalid history ids and/or when checking the smt object.

Reply:
True, it'll fail either way, so this is fine. I was just thinking to terminate execution of the view early, but it probably isn't much of a performance savings.